### PR TITLE
DAOS-4672 Client: Agent failure with no network devices

### PR DESCRIPTION
**Pull request by jbrosenz**
Resolves an error that occurs when the agent detects
    no network devices and generates a divide by 0 in the
    load balancer.

    The response cache now includes a default response
    that is always available.  The default response contains
    no network device and domain information.  This response
    is given when there is no NUMA node specific response
    available.

Signed-off-by: Joel Rosenzweig <joel.b.rosenzweig@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
I also observed this issue, and the PR fixes it for me.

**Comment by jbrosenz** [comment]
I'm trying to get a clean run through CI.  It would be a good idea to get review feedback at this time in case you need any significant changes.  Thank you.

File: src/control/cmd/agent/infocache.go [hunk 1]
```diff
@@ -50,6 +51,8 @@ type attachInfoCache struct {
 	initialized atm.Bool
 	// maps NUMA affinity and device index to a response
 	numaDeviceMarshResp map[int]map[int][]byte
+	// response given when there is no numa or device data
+	defaultResp []byte
```
Comment (mjmac) [inline src/control/cmd/agent/infocache.go:55]: I'm not super crazy about this... Wouldn't it be better to look at the length of the marshaled response map after the device scan loop and if it's empty, just create an entry for numa node 0, device 0? It's kind of a kludge either way, but this way adds more code that has to be scrutinized/maintained.

**Comment by jbrosenz** [comment]
I resubmitted this after making a clean tree/build in case I was having cruft in the build.  I couldn't get past the Clang failure in this submission, but the new PR made it through.  I'll close this one in favor of the new one.  https://github.com/daos-stack/daos/pull/2592

File: src/control/cmd/agent/infocache.go (hunk 1 above)
Comment (jbrosenz) [reply src/control/cmd/agent/infocache.go:55]: I can do that.  I'll submit the change in https://github.com/daos-stack/daos/pull/2592

**Comment by ashleypittman** [comment]
You need to delete the Branch after closing the PR.

**Comment by jbrosenz** [comment]
> You need to delete the Branch after closing the PR.

I will be closing it today.  I had some more thoughts to add to Michael's comment and wasn't ready to delete the branch until I either addressed it here, or moved the comment over to the new PR.

**Pull request closed without merging**